File: 4258-construct-uniform-parity-array-ii/construct-uniform-parity-array-ii.c
bool uniformArray(int* nums1, int nums1Size) {
    int even = nums1[0], min = nums1[0];
    for(int i = 1; i < nums1Size; i++) {
        min = fmin(min, nums1[i]);
        even |= nums1[i];
    }
    return (min & 1) || !(even & 1);
}

File: 4258-construct-uniform-parity-array-ii/construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int even = nums1[0], min = nums1[0];
        for(int i = 1; i < nums1.length; i++) {
            min = Math.min(min, nums1[i]);
            even |= nums1[i];
        }
        return (min & 1) == 1 || (even & 1) == 0;
    }
}